templates/CMakeLists.txt.template

```diff
@@ -145,9 +145,9 @@
   set(gRPC_CORE_VERSION     "${settings.core_version}")
   set(gRPC_CORE_SOVERSION   "${settings.core_version.major}")
   set(gRPC_CPP_VERSION      "${settings.cpp_version}")
-  set(gRPC_CPP_SOVERSION    "${settings.cpp_version.major}")
+  set(gRPC_CPP_SOVERSION    "${settings.core_version.major}")
```

Comment:
the SO version of grpc++ is actually supposed to be 1.x.  the grpc library (=C core) is versioned differently (and is currently at version 15).  We'd have to have a really strong reason to change that since the change you're proposing is intuitively incorrect.

Reply:
Good point. The intention was just to make a differentiation on the SONAME. I've adapted as you suggest uggest to set `"${settings.cpp_version.major}.${settings.cpp_version.major}"`. It would look as below:

```
lrwxrwxrwx  1       17 Mar  5 10:39 libgrpc++.so -> libgrpc++.so.1.37
lrwxrwxrwx  1       23 Mar  5 10:39 libgrpc++.so.1.37 -> libgrpc++.so.1.37.0-dev
-rwxrwxr-x  1  2789096 Mar  5 10:39 libgrpc++.so.1.37.0-dev
lrwxrwxrwx  1       22 Mar  5 10:40 libgrpc++_alts.so -> libgrpc++_alts.so.1.37
lrwxrwxrwx  1       28 Mar  5 10:40 libgrpc++_alts.so.1.37 -> libgrpc++_alts.so.1.37.0-dev
-rwxrwxr-x  1   144864 Mar  5 10:40 libgrpc++_alts.so.1.37.0-dev
lrwxrwxrwx  1       31 Mar  5 10:39 libgrpc++_error_details.so -> libgrpc++_error_details.so.1.37
lrwxrwxrwx  1       37 Mar  5 10:39 libgrpc++_error_details.so.1.37 -> libgrpc++_error_details.so.1.37.0-dev
-rwxrwxr-x  1     7416 Mar  5 10:39 libgrpc++_error_details.so.1.37.0-dev
lrwxrwxrwx  1       28 Mar  5 10:40 libgrpc++_reflection.so -> libgrpc++_reflection.so.1.37
lrwxrwxrwx  1       34 Mar  5 10:40 libgrpc++_reflection.so.1.37 -> libgrpc++_reflection.so.1.37.0-dev
-rwxrwxr-x  1  1120672 Mar  5 10:40 libgrpc++_reflection.so.1.37.0-dev
lrwxrwxrwx  1       26 Mar  5 10:39 libgrpc++_unsecure.so -> libgrpc++_unsecure.so.1.37
lrwxrwxrwx  1       32 Mar  5 10:39 libgrpc++_unsecure.so.1.37 -> libgrpc++_unsecure.so.1.37.0-dev
-rwxrwxr-x  1  2418208 Mar  5 10:39 libgrpc++_unsecure.so.1.37.0-dev
```

Thanks

Reply:
And same way, checking the artifacts:

```bash
$ objdump -p libgrpc++.so | grep SONAME
  SONAME               libgrpc++.so.1.37
$ readelf -a libgrpc++_reflection.so | grep NEED
  [ 6] .gnu.version_r    VERNEED          0000000000058200  00058200
 0x0000000000000001 (NEEDED)             Shared library: [libgrpc++.so.1.37]
 0x0000000000000001 (NEEDED)             Shared library: [libprotobuf.so.3.15.2.0]
 0x0000000000000001 (NEEDED)             Shared library: [libstdc++.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libgcc_s.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
 0x000000006ffffffe (VERNEED)            0x58200
 0x000000006fffffff (VERNEEDNUM)         3
```

Thanks

Reply:
Rebased to latest master. @jtattermusch please let me know if there's a better approach to address this issue. Perhaps putting the whole version would be even better, fe see what boost does. A proper handling of SONAME would be really useful for downstreams and similar situations.

Thanks